1207 create a demo shell script that illustrates integration with CircleCI

AlexAndrosov opened the pull request:
ref: microservices-platform/microservices-assessment-platform#1207

-Added .env file in gitignore file
-Added script which creates new service

mxzpytlk committed:
Issue microservices-platform/microservices-assesment-platform#1207 GraphQL: create a demo shell script that illustrates integration with CircleCI (for example): script
-Added .env file in gitignore file
-Added script which creates new service

cer commented on create-service-and-ensure-existence.sh:
```diff
@@ -0,0 +1,18 @@
+service=$3
+
+curl -X POST "$API_ENDPOINT" \
+-u $USER_LOGIN:$USER_PASSWORD \
```
These need to prefixed with MSAT_ - I don't think they are 'login' and 'password' - api access....

@andrevinsky

cer commented on create-service-and-ensure-existence.sh:
```diff
@@ -0,0 +1,18 @@
+organization=$1
```
THis should start with `#! /bin/bash -e`

`-e` is important so that it will fail on error.

cer commented on create-service-and-ensure-existence.sh:
```diff
@@ -0,0 +1,18 @@
+organization=$1
```
Are these parameters mandatory?
if so they should be `${1?}`

AlexAndrosov replied:
No. It is optional.

mxzpytlk committed:
Issue microservices-platform/microservices-assesment-platform#1207 GraphQL: create a demo shell script that illustrates integration with CircleCI (for example): script
-Added curl which set github repository metadata to service

dartandrevinsky commented on create-service-and-ensure-existence.sh:
```diff
@@ -0,0 +1,39 @@
+#! /bin/bash -e
```
I think the filename should be: `ensure-service-and-update-repo-metadata.sh`

dartandrevinsky replied:
Also, please add a section to README to describe how to use this script. E.g. from within `circleci.yml` config.

cer replied:
> I think the filename should be: `ensure-service-and-update-repo-metadata.sh`

msat-set-service-repo-url-metadata.sh

mxzpytlk committed:
Issue microservices-platform/microservices-assesment-platform#1207 GraphQL: create a demo shell script that illustrates integration with CircleCI (for example): rename script
-Renamed script name from "create-service-and-ensure-existence.sh" to "ensure-service-and-update-github-repo.sh"

mxzpytlk committed:
Issue microservices-platform/microservices-assesment-platform#1207 GraphQL: create a demo shell script that illustrates integration with CircleCI (for example): rename script
-Renamed script name from "ensure-service-and-update-github-repo.sh" to "msat-set-service-repo-url-metadata.sh"
-Added description "How to build service"

cer commented on docs/build-service.adoc:
```diff
@@ -0,0 +1,10 @@
+----
+
+It will create service if it does not exist and set github repository as metadata property.
+To run this script environment variables MSAT_API_ENDPOINT, MSAT_USER_LOGIN and MSAT_USER_PASSWORD must be set.
```
`MSAT_API_ENDPOINT` should be an optional command line argument that defaults to the production endpoint.

cer replied:
The MSAT API does not have a 'user login` and `user password`.

It has an `Access Key ID` and `Secret Access Key`

cer commented on msat-set-service-repo-url-metadata.sh:
```diff
@@ -0,0 +1,39 @@
+organization=$3
+
+curl -X POST "$MSAT_API_ENDPOINT" \
+-u $MSAT_USER_LOGIN:$MSAT_USER_PASSWORD \
```
Required env. vars should be `${X?}`

cer commented on docs/build-service.adoc:
```diff
@@ -0,0 +1,10 @@
+== Build service
```
This does not 'build a service' - it sets the github url metadata property

cer commented on msat-set-service-repo-url-metadata.sh:
```diff
@@ -0,0 +1,39 @@
+  }
+'
+
+github_repo=`git remote -v | grep -o "origin\s.*\s(fetch)" | awk '{ print $2 }'`
```
Simpler: git remote get-url origin

mxzpytlk committed:
Issue microservices-platform/microservices-assesment-platform#1207 GraphQL: create a demo shell script that illustrates integration with CircleCI (for example): fix
-Renamed "build-service.adoc" to "set-service-github-url.adoc"
-Added api-endpoint as an optional variable
-Renamed environment variable MSAT_USER_LOGIN to MSAT_ACCESS_KEY_ID
-Renamed environment variable MSAT_USER_PASSWORD to MSAT_SECRET_ACCESS_KEY

cer commented on msat-set-service-repo-url-metadata.sh:
```diff
@@ -0,0 +1,58 @@
+	shift
+done
+
+if [ -z $api_endpoint ]
```
Eliminate this conditional logic by moving `api_endpoint="https://platform.microservices.io/graphql"` to before the while

mxzpytlk committed:
Issue microservices-platform/microservices-assesment-platform#1207 GraphQL: create a demo shell script that illustrates integration with CircleCI (for example): refactoring
-Moved api_endpoint definition to before the while
